plumbing/util.go

```diff
@@ -48,6 +55,90 @@ func ExecToStdout(cmdString string, targetDir string) error {
 	return ExecToStdoutWithEnv(cmdString, targetDir, os.Environ())
 }
 
+func ExecToLog(cmdString string, targetDir string, logPath string) error {
+	cmdArgs, err := shlex.Split(cmdString)
+	if err != nil {
+		return fmt.Errorf("Can't parse command string '%s': %v", cmdString, err)
+	}
+
+	logfile, err := os.OpenFile(logPath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return fmt.Errorf("Couldn't open log file %s: %v", logPath, err)
+	}
+
+	defer logfile.Close()
+
+	cmd := exec.Command(cmdArgs[0], cmdArgs[1:len(cmdArgs)]...)
+	cmd.Dir = targetDir
+	cmd.Stdout = logfile
+	cmd.Stdin = os.Stdin
+	cmd.Stderr = logfile
+
+	err = cmd.Run()
+
+	if err != nil {
+		return fmt.Errorf("Command '%s' failed to run with error -- see log for information: %s", cmdString, logPath)
+	}
+
+	return nil
+
+}
+
+func ExecSilently(cmdString string, targetDir string) error {
+	cmdArgs, err := shlex.Split(cmdString)
+	if err != nil {
+		return fmt.Errorf("Can't parse command string '%s': %v", cmdString, err)
+	}
+
+	cmd := exec.Command(cmdArgs[0], cmdArgs[1:len(cmdArgs)]...)
+	cmd.Dir = targetDir
+	cmd.Stdout = ioutil.Discard
+	cmd.Stdin = os.Stdin
+	cmd.Stderr = ioutil.Discard
+
+	err = cmd.Run()
+
+	if err != nil {
+		return fmt.Errorf("Command '%s' failed to run with error %v", cmdString, err)
+	}
+
+	return nil
+
+}
+
+func ExecToLogWithProgressDots(cmdString string, targetDir string, logPath string) error {
+	stoppedchan := make(chan struct{})
+	dotchan := make(chan int)
+	defer close(stoppedchan)
+
+	go func() {
+		for {
+			select {
+			default:
+				dotchan <- 1
+				time.Sleep(3 * time.Second)
+			case <-stoppedchan:
+				return
+			}
+		}
+	}()
+
+	go func() {
+		for {
+			select {
+			default:
+			case <-dotchan:
+				fmt.Printf(".")
```

Comment:
It may look better if we only show 5 dots at max, after 5 we should delete all dots and start printing them again, inplace.

Reply:
I thought about that but, in particular with remote logs being sent back we don't have control over the terminal directly and so it makes the streamed logs look funny. We could increase the time between dots maybe; thoughts?

Reply:
Yeah, maybe 10 seconds